tools/manifest/sourcefile.py

```diff
@@ -674,6 +674,32 @@ def has_testdriver(self):
             return None
         return bool(self.testdriver_nodes)
 
+    @cached_property
+    def quic_nodes(self):
+        # type: () -> List[ElementTree.Element]
+        """List of ElementTree Elements corresponding to nodes in a test that
+        specify whether it needs QUIC server."""
+        assert self.root is not None
+        return self.root.findall(".//{http://www.w3.org/1999/xhtml}meta[@name='quic']")
+
+    @cached_property
+    def quic(self):
+        # type: () -> Optional[bool]
+        """Boolean indicating whether a test requires QUIC server"""
+        if self.script_metadata:
+            if any(m == (b"quic", b"true") for m in self.script_metadata):
+                return True
+
+        if self.root is None:
+            return None
+
+        if self.quic_nodes:
+            quic_str = self.quic_nodes[0].attrib.get("content", None)  # type: Optional[Text]
+            if quic_str and quic_str.lower() == "true":
```

Comment:
Optional nit; could we default to a "false" string instead, turning the second line into just `quic_str.lower() == "true"` without the null check?

Reply:
Again this is largely for consistency. This class differentiates missing (`None`) and falsy values (see `has_testdriver`). And returning `None` actually saves us some bytes in MANIFEST: we'll omit the field instead of storing `quic: false` everywhere.

Reply:
I was referring to the second argument of `get`, which I assume is a default value if the item isn't found (I'm unclear on the type of `attrib`). This actual function should return True or None.

Reply:
Aha, the more important difference is whether to differentiate false and null. My original implementation had three possible return values. And thinking again, there's really no need to return false ever.